### PR TITLE
Some touchups / nitpicks before release

### PR DESCRIPTION
**Pull request by tbekolay**
Heyo, took a look through this repo and it's looking good! This PR includes a few minor touchups and nitpick commits.

I did two non-minor things that you can take or leave:

653aa52d5ee3db555300789a768ce2d20dc6d877 makes Nengo DL compatible with Python 2.7 (it works on my machine, anyhow). Of course, this means that some things are not as nice as if it was Python 3 only, so if you want to keep it Python 3 only that's cool too!

5d40504f5cdb9eb9d0e32 renames `TestSimulator` because of some pytest warnings. The new name, `SimulatorWithTestingDefaults`, is pretty heinous, so I would recommend you come up with a better name than that (I just wasn't super sure what to call it myself).

### CODE REVIEW

File: nengo_deeplearning/utils.py [hunk 1]
```diff
@@ -292,8 +292,11 @@ def minibatch_generator(inputs, targets, minibatch_size, shuffle=True,
     rng : :class:`~numpy:numpy.random.RandomState`, optional
         random number generator
 
-    Yields
-    ------
+    Returns
+    -------
+
+    A generator that yeilds:
```
Comment (tbekolay) [inline nengo_deeplearning/utils.py]: Hm, sphinx raised a warning for me that `numpydoc` didn't understand it. Do you not get a warning? Maybe I'm using an older version of `numpydoc`?

**Comment by tbekolay** [comment]
Yeah, eventually all Nengo project should drop 2.7 I think. :+1:

After a very professional number of fixup commits, I got the documentation building through TravisCI; see https://nengo.github.io/nengo_deeplearning/. I had it build on this branch in one commit, but in general, every branch will build the documentation, but the docs will only be pushed to the `gh-pages` branch if the TravisCI build is started from the master branch.

File: nengo_deeplearning/utils.py (hunk 1 above)
Comment (drasmuss) [inline nengo_deeplearning/utils.py]: Apparently it was added in 0.6 https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt

Comment (tbekolay) [inline nengo_deeplearning/utils.py]: So it was! I upgraded and the warning goes away, so I'll push a commit reverting this change.

**Comment by tbekolay** [comment]
I'm not sure where the discussion on the `SimulatorWithTestingDefaults` name went, but `TestingSimulator` won't work as pytest checks if the name starts with `Test` so that'll still get picked up. One option would be to keep the name as `Simulator`; since it'll live in `nengo_deeplearning.tests` it's differentiated by the module (`nengo_deeplearning.Simulator` vs `nengo_deeplearning.tests.Simulator`). To be extra clear that they're different, we could add an underscore maybe (`nengo_deeplearning.tests._Simulator`)?

I noticed that tests were failing sometimes despite the Nengo seed being set, and figured that it was likely the tensforflow optimization process being non-deterministic. I pushed 26f6b1aa9e9690c that adds a `seed` argument to `Simulator.train`, added a seed to those calls, and now I get the same RMSE every time I run those tests.

I realize now, though, that `Simulator` already has a seed that it uses for processes. Should we re-use that seed to set the tensorflow seed when training? Or should we not mess with the tensorflow seed at all, and instead explicitly call `tf.set_random_seed` in these tests?

**Comment by drasmuss** [comment]
Thanks for tracking that down.  I'm really curious where the randomness is coming from, since gradient descent optimization should be deterministic.  But we can just chalk that up to more tensorflow weirdness for now.

I like the idea of using the Simulator seed, seems like an intuitive way to do it.  Added a fixup commit to do it that way.  I think more long-term it'd be better not to mess with the global tensorflow seed, so hopefully I'll be able to figure out where the rng is coming from and fix that at the source.  But this is a good fix for now.

I also committed an idea for the `TestSimulator` naming, where I went back to `TestSimulator` but then changed the pytest settings so it doesn't try to collect classes.  Not ideal if we ever did want to have a test class, but I've never actually wanted to do that so it seems an okay, if somewhat hacky, solution.  How does that sound?

**Comment by tbekolay** [comment]
Unfortunately we do have some test classes in the Nengo test suite, so this causes fewer tests to be collected when doing `pytest --pyargs nengo`.

**Comment by drasmuss** [comment]
Two more fixup commits, to use `tests.Simulator` and slightly simplify `travis.yml`.  I think we've reached peak fixup.

File: CHANGES.rst [hunk 2]
```diff
@@ -19,7 +19,12 @@ Release History
    - Removed
    - Fixed
 
-0.1.0 (unreleased)
+0.1.0 (12/06/2016)
```
Comment (tbekolay) [inline CHANGES.rst]: In other Nengo projects we use the date format Month DD, YYYY (e.g. December 6, 2016) as it tends to be easier to read when scanning down a list of releases... dunno how important consistency is there, but might as well be consistent I guess?

File: CHANGES.rst [hunk 3]
```diff
@@ -19,7 +19,12 @@ Release History
...
 ==================
 
-Initial release
+Initial release of Lasagne-based NengoDL
+
+0.2.0 (unreleased)
```
Comment (tbekolay) [inline CHANGES.rst]: We also do reverse chronological order for releases elsewhere (so newest to oldest). I think I just copied that organization from other projects, but people tend to want to know what's new in this release mostly, then ignore the rest, so there's some reasoning behind it.
